File: app/src/main/java/com/example/myfakequotesapp/Room/RepositoryTasks.java
package com.example.myfakequotesapp.Room;

import androidx.lifecycle.LiveData;

import com.example.myfakequotesapp.Model.Quote;

import java.util.List;

public interface RepositoryTasks {
    <T extends Quote> LiveData<List<T>> getAllQuotes();
    <T extends Quote> Quote getQuoteByID(int quoteId);
    <T extends Quote> void addQuote(T Quote);
    <T extends Quote> void deleteQuote(T Quote);
    <T extends Quote> void updateQuote(T Quote);
}

File: app/src/main/java/com/example/myfakequotesapp/Room/RepositoryTasks.kt
package com.example.myfakequotesapp.Room

import androidx.lifecycle.LiveData
import com.example.myfakequotesapp.Model.Quote
import org.jetbrains.annotations.NotNull

interface RepositoryTasks {
    @NotNull
    fun <T : Quote?> getAllQuotes(): LiveData<List<T?>?>?
    fun <T : Quote?> getQuoteByID(quoteId: Int): Quote?
    fun <T : Quote?> addQuote(Quote: T)
    fun <T : Quote?> deleteQuote(Quote: T)
    fun <T : Quote?> updateQuote(Quote: T?)
    //fun updateQuote(quote: Quote?)
}